unistd/dir.c

```diff
@@ -229,6 +231,12 @@ static int _resolve_abspath(char *path, char *result, int resolve_last_symlink,
 			}
 		}
 
+		if ((is_leaf != 0) && (resolve_last_symlink == 0)) {
+			/* TODO: in this case it's more efficient to check if file exists
```

Comment:
1. Checking if path exists and is a directory before call to `_readlink_abs` may be slightly more efficient in the case when `(is_leaf != 0) && (resolve_last_symlink == 0)` and the symlink exists. With this code the symlink's contents will be read even though it's not necessary. It's probably not a big performance concern.
2. You're right, when doing `lstat` on `/path/to/symlink/`, if `symlink` exists and is a symlink then the call will succeed despite the trailing slash. I think in this case we need to check `if (ends_with_slash != 0)` and exit with `errno == -ENOTDIR`.

Reply:
> 1. Checking if path exists and is a directory before call to `_readlink_abs` may be slightly more efficient in the case when `(is_leaf != 0) && (resolve_last_symlink == 0)` and the symlink exists. With this code the symlink's contents will be read even though it's not necessary. It's probably not a big performance concern.

still, this should be **instead of**, not **before** as in the mentioned case you don't want to call `_readlink_abs` at all (if we really want to improve performance).  I think it's not worth the additional code complexity (not a hot path), so IMHO `TODO` should be removed

Regarding (2) -> maybe add a test for this particular "corner case" to avoid regressions in the future (this code is hard enough to understand with all special cases ;))